### PR TITLE
Adding Part_ Template parity

### PR DESCRIPTION
**Pull request by harshit7962**
## Description
The following changes creates the minimal parity of **PART_ templates** in the Fluent styles wrt Aero2 Styles. Stylings of a few controls in Fluent missed having PART_ templates that would likely cause regressions when Fluent styles are made default. The following controls were modified to include the PART_ template:
1. ComboBox
2. MenuItem
3. Frame and
4. DatePicker
5. DatePickerTextBox

## Testing
Local Build Pass
Sample Application Testing
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/wpf/pull/9398)

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/src/Themes/PresentationFramework.Fluent/Styles/Frame.xaml [hunk 1]
```diff
@@ -7,13 +7,157 @@
...
 <ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
 
+    <ControlTemplate x:Key="FrameNavChromeTemplateKey"
+                     TargetType="{x:Type Frame}">
+        <Border Background="{TemplateBinding Background}"
+                BorderBrush="{TemplateBinding BorderBrush}"
+                BorderThickness="{TemplateBinding BorderThickness}"
+                Padding="{TemplateBinding Padding}">
+            <DockPanel>
+                <Grid Background="{StaticResource NavigationWindowNavigationChromeBackground}"
```
Comment (dipeshmsft) [inline src/Microsoft.DotNet.Wpf/src/Themes/PresentationFramework.Fluent/Styles/Frame.xaml:17]: Check if these static resources like NavigationWindowNavigationChromeBackground , NavigationWindowForwardButtonStyle etc are defined.

Comment (harshit7962) [reply src/Microsoft.DotNet.Wpf/src/Themes/PresentationFramework.Fluent/Styles/Frame.xaml:17]: As discussed these resources are present in `navigationwindow` control which was added as part of the Missing controls [PR](https://github.com/dotnet/wpf/pull/9171).

**Review by dipeshmsft — DISMISSED** [review]

**Review by sabareesanms — CHANGES_REQUESTED** [review]

File: src/Microsoft.DotNet.Wpf/src/Themes/PresentationFramework.Fluent/Styles/DatePicker.xaml [hunk 2]
```diff
@@ -46,9 +46,18 @@
...
-                    </Border>
+                    <Grid>
+                        <Border>
+                            <ContentControl x:Name="PART_Watermark"
+                                Opacity="0"
+                                Focusable="False"
+                                IsHitTestVisible="False"
+                                Padding="2"/>
+                        </Border>
+                        <Border Margin="{TemplateBinding Margin}">
```
[the remark this answers is not in the document]
Comment (harshit7962) [reply src/Microsoft.DotNet.Wpf/src/Themes/PresentationFramework.Fluent/Styles/DatePicker.xaml:57]: Addressed. Changed the order of controls.

File: src/Microsoft.DotNet.Wpf/src/Themes/PresentationFramework.Fluent/Styles/DatePicker.xaml [hunk 3]
```diff
@@ -46,9 +46,18 @@
...
-                    <Border Margin="{TemplateBinding Margin}">
-                        <Decorator x:Name="PART_ContentHost" Margin="{TemplateBinding Padding}" />
-                    </Border>
+                    <Grid>
+                        <Border>
+                            <ContentControl x:Name="PART_Watermark"
+                                Opacity="0"
+                                Focusable="False"
+                                IsHitTestVisible="False"
+                                Padding="2"/>
```
[the remark this answers is not in the document]
Comment (harshit7962) [reply src/Microsoft.DotNet.Wpf/src/Themes/PresentationFramework.Fluent/Styles/DatePicker.xaml:55]: Addressed. Removed padding and binded padding to margin instead.

**Review by sabareesanms — APPROVED** [review]

**Pull request merged**